### ISSUE TITLE
Use a build config to build the xdmod containers on the openshift cluster

### ISSUE THREAD

**Issue by rob-baron** [CCI-MOC/xdmod-cntr#29]
OpenShift 4.x will build a container using a docker file from a referenced GitHub repository.  This is more convenient than using a local instance of either podman or docker to build the container and push the container to docker hub, then download it when the project is deployed on openshift.

**Comment by rob-baron** [comment]
This now works in the cop-staging cluster after the internal registry was opened.  test and push the build config and image stream after the nerc-openshift cluster is configured.

**Comment by rob-baron** [comment]
1) it is a nice to have as it makes the install a bit more self contained.
2) I am just waiting until the NERC is setup to do this sort of build.
3) as it is just yaml and I have tested it on staging this has maybe an hour left - have to change the yaml to refer to the xdmod container that is built by this process as opposed to the one that I'm uploading to docker hub.

**Comment by joachimweyl** [comment]
@rob-baron just to clarify, specifically which NERC setup are we waiting on for this?

**Comment by joachimweyl** [comment]
@rob-baron from your comment in standup today it sounds like this is done?

**Comment by rob-baron** [comment]
First of all this is done.

To Recap:

We had figured out that code ready containers exposes the internal registry by default, where as OpenShift does not by default.  So lars had to expose the internal registry on ocp-staging to allow building docker files there (of which I commented on farther up in the ticket).  

In one meeting we had with the NERC folks, they agreed to expose the internal registry.  After a few days, when I realized that this was not done, I mentioned it a slack thread about exposing the internal registry.  I tried it again after a couple of days and it was done.

This was low hanging fruit, but does allow me to reduce my dependency on docker desktop.
